training/tf/tfprocess.py

```diff
@@ -65,49 +71,63 @@ def optimistic_restore(session, save_file, graph=tf.get_default_graph()):
     opt_saver = tf.train.Saver(restore_vars)
     opt_saver.restore(session, save_file)
 
+
 # Class holding statistics
 class Stats:
     def __init__(self):
         self.s = {}
+
     def add(self, stat_dict):
-        for (k,v) in stat_dict.items():
+        for (k, v) in stat_dict.items():
             if k not in self.s:
                 self.s[k] = []
             self.s[k].append(v)
+
     def n(self, name):
         return len(self.s[name] or [])
+
     def mean(self, name):
         return np.mean(self.s[name] or [0])
+
     def stddev_mean(self, name):
         # standard deviation in the sample mean.
         return math.sqrt(
             np.var(self.s[name] or [0]) / max(0.0001, (len(self.s[name]) - 1)))
+
     def str(self):
         return ', '.join(
-            ["{}={:g}".format(k, np.mean(v or [0])) for k,v in self.s.items()])
+            ["{}={:g}".format(k, np.mean(v or [0])) for k, v in self.s.items()])
+
     def clear(self):
         self.s = {}
+
     def summaries(self, tags):
         return [tf.Summary.Value(
-            tag=k, simple_value=self.mean(v)) for k,v in tags.items()]
+            tag=k, simple_value=self.mean(v)) for k, v in tags.items()]
+
 
 # Simple timer
 class Timer:
     def __init__(self):
         self.last = time.time()
+
     def elapsed(self):
         # Return time since last call to 'elapsed()'
         t = time.time()
         e = t - self.last
         self.last = t
         return e
 
+
 class TFProcess:
     def __init__(self):
         # Network structure
         self.RESIDUAL_FILTERS = 128
         self.RESIDUAL_BLOCKS = 6
 
+        # Set number of GPUs for training
+        self.gpus_num = 2
```

Comment:
The default should be 1, so as not to break existing scripts et al.

Reply:
Yes, this is a change for my test, I will fix it.